src/js/components/Video.js

```diff
@@ -21,24 +15,23 @@ export default class Video extends Component {
   constructor () {
     super();
 
-    this._onResponsive = this._onResponsive.bind(this);
-    this._onPlaying = this._onPlaying.bind(this);
-    this._onPause = this._onPause.bind(this);
-    this._onEnded = this._onEnded.bind(this);
-    this._onClickControl = this._onClickControl.bind(this);
+    this._hasPlayed = false;
+    this._play = this._play.bind(this);
+    this._pause = this._pause.bind(this);
+    this._togglePlay = this._togglePlay.bind(this);
+    this._toggleMute = this._toggleMute.bind(this);
+    this._seek = this._seek.bind(this);
+    this._mute = this._mute.bind(this);
+    this._unmute = this._unmute.bind(this);
+    this._fullscreen = this._fullscreen.bind(this);
     this._onMouseMove = this._onMouseMove.bind(this);
-    this._onClickChapter = this._onClickChapter.bind(this);
-    this._onFullScreen = this._onFullScreen.bind(this);
 
-    this.state = { playing: false, progress: 0, iconSize: 'large' };
+    this.state = {};
   }
 
-  componentDidMount () {
-    this._responsive = Responsive.start(this._onResponsive);
-    let video = this.refs.video;
-    video.addEventListener('playing', this._onPlaying);
-    video.addEventListener('pause', this._onPause);
-    video.addEventListener('ended', this._onEnded);
+  componentWillMount () {
+    this._update = throttle(this._update.bind(this), 100).bind(this);
```

Comment:
@alansouzati I haven't tried it yet but from looking at the logic, it does not seem to work for this case. We want to throttle the call here from happening too often within 100ms but the function will still get called every 100ms. As with debounce logic, it looks to me like the function will not get called at all if the event keeps getting triggered within 100ms. 

With that said, in this case, there are multiple Video events (https://github.com/grommet/grommet/pull/758/files#diff-494d83d33bab8a72199104e6107d5b18R49) we are listening, the `_update()` function might not get called for a good second or two if we were to use the debounce logic.